doc/smart_contract/nfr_ctrt_v2.md

```diff
@@ -0,0 +1,505 @@
+# NFT Contract V2
+
+- [NFT Contract V2](#nft-contract-v2)
+  - [Introduction](#introduction)
+  - [Usage with Javascript SDK](#usage-with-js-sdk)
+    - [Registration](#registration)
+    - [From Existing Contract](#from-existing-contract)
+    - [Querying](#querying)
+      - [Issuer](#issuer)
+      - [Maker](#maker)
+      - [Regulator](#regulator)
+      - [Unit](#unit)
+      - [Is user in the list](#is-user-in-the-list)
+      - [Is contract in the list](#is-contract-in-the-list)
+    - [Actions](#actions)
+      - [Issue](#issue)
+      - [Send](#send)
+      - [Transfer](#transfer)
+      - [Deposit](#deposit)
+      - [Withdraw](#withdraw)
+      - [Supersede](#supersede)
+      - [Add/remove a user from the list](#addremove-a-user-from-the-list)
+      - [Add/remove a contract from the list](#addremove-a-contract-from-the-list)
+
+## Introduction
+
+NFT contract V2 adds additional whitelist/blacklist regulation feature upon [NFT contract V1](./nft_ctrt.md).
+
+For the whitelist flavor, only users & contracts included in the list can interact with the NFT contract instance.
+
+For the blacklist flavor, only users & contracts excluded from the list can interact with the NFT contract instance.
+
+## Usage with JS SDK
+
+Examples of NFT contract V2 with whitelist are shown below. The usage of the blacklist one is very similar.
+
+### Registration
+
+```Javascript
+import * as nftv2 from './src/contract/nft_ctrt_v2.js';
+
+// acnt: Account
+
+// Register a new NFT contract
+const nc = await nftv2.NFTCtrtV2Whitelist.register(acnt);
+console.log(nc.ctrtId); // print the id of the newly registered contract
+```
+
+Example output
+
+```
+CtrtID { data: 'CF46ab6o5HTfyLwgyBhwrhkEmLxbaLkSJ8a' }
+```
+
+### From Existing Contract
+
+```javascript
+import * as nftv2 from './src/contract/nft_ctrt_v2.js';
+
+// acnt: Account
+
+const nc_id = 'CF46ab6o5HTfyLwgyBhwrhkEmLxbaLkSJ8a';
+const nc = nftv2.NFTCtrtV2Whitelist(nc_id, ch);
```

Comment:
from line 60 to 63: 1. add `// ch: Chain` 2. use `ncId` instead of` nc_id`(we prefer camel case), should check the other docs and update them.